### PR TITLE
feat-fix: improve usedRVersion to return after a reasonable timeout

### PR DESCRIPTION
**Pull request by Ellpeck**
This improvement/fix also includes an additional property for `CollectorTimeout` that allows skipping the rejection when the timeout is reached.

### CODE REVIEW

File: src/r-bridge/shell.ts [hunk 1]
```diff
@@ -53,7 +57,8 @@ export const DEFAULT_OUTPUT_COLLECTOR_CONFIGURATION: OutputCollectorConfiguratio
 	postamble: `🐧${'-'.repeat(5)}🐧`,
 	timeout:   {
 		ms:             750_000,
-		resetOnNewData: true
+		resetOnNewData: true,
+		reject:         true
```
Comment (EagleoutIce) [inline src/r-bridge/shell.ts:61]: i am not such a big fan of a `reject` flag. how about a more general purpose `onError` handler?

Comment (Ellpeck) [reply src/r-bridge/shell.ts:61]: fiiine

File: src/r-bridge/shell.ts [hunk 2]
```diff
@@ -21,6 +22,10 @@ export interface CollectorTimeout extends MergeableRecord {
    * if true, the timeout will reset whenever we receive new data
    */
 	resetOnNewData: boolean
+	/**
+	 * invoked when the timeout is reached. If not set, the promise will be rejected with an error.
+	 */
+	onTimeout?:     (resolve: (value: AsyncOrSync<string[]>) => void, reject: (value: AsyncOrSync<string[]>) => void, partialResult: string[]) => void
```
Comment (EagleoutIce) [inline src/r-bridge/shell.ts:28]: i expected a handler for the error case in general but that is fine :)

**Review by EagleoutIce — APPROVED** [review]

**Comment by EagleoutIce** [comment]
_This pull request is included in v2.0.0 (see [Release v2.0.0 (Dataflow v2)](https://github.com/Code-Inspect/flowr/releases/tag/v2.0.0))._

**Pull request merged**